### PR TITLE
DAC Filter A11y Compliance

### LINKED ISSUE #3061: [a11y fix] Django multi-select filter
**Description:**
Follow on to #3037

Implements changes to correct a11y of the DAC multiselect component:
- Label changes to make the broader actions clearer, "All" --> "Show all", "filter" --> "Filter to selection(s)". 
- Adds aria-mutliselectable="true" to the select component and aria-checked="true" to selected items.

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [x] Labels have been changed
   - [x] "All" reads as "Show all" 
   - [x] "filter" reads as "Filter to selection(s)"
- [x] aria-mutliselectable="true" is on select component
- [x] aria-checked="true" is on actively selected items
- [x] Testing Checklist has been run and all tests pass
- [x] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [x] Implement label changes
- [x] Implement aria changes
- [x] a11y test
- [x] Tag in design for raft a11y review
- [x] Run Testing Checklist and confirm all tests pass

### PR DESCRIPTION
**Pull request by elipe17**
## Summary of Changes
- Removed third party dependency
- Added new template and supporting code for multi select
- Made style and a11y changes to the new template
Pull request closes #3061 

## How to Test
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.
2. Optionally submit a datafile or two
3. Go to one of the search indexes tabs and verify the filter meets ACs

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Labels have been changed
   - [x] "All" reads as "Show all" 
   - [x] ~~"filter" reads as "Filter to selection(s)"~~
- [x] aria-mutliselectable="true" is on select component
- [x] aria-checked="true" is on actively selected items
- [x] Testing Checklist has been run and all tests pass
- [x] README is updated, if necessary

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [x] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [x] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by andrew-jameson — APPROVED** [review]
LGTM

File: tdrs-backend/tdpservice/search_indexes/admin/multiselect_filter.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
+"""File containing multiselect filter classes and mixins."""
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/search_indexes/admin/multiselect_filter.py:1]: are these being copied from the removed library for customization?

**Review by jtimpe — APPROVED** [review]
Small question for clarification. Otherwise looks/works as expected and i see the `aria` tags in the DOM. Good work!

File: tdrs-backend/tdpservice/search_indexes/admin/multiselect_filter.py (hunk 1 above)
Comment (elipe17) [reply tdrs-backend/tdpservice/search_indexes/admin/multiselect_filter.py:1]: Correct, I pulled them from the library and modified them to meet our needs

**Comment by elipe17** [comment]
As a note, #3110 will need to be worked as follow on to this to resolve an annoying issue with 508/TDP filters. Does not break functionality, but it does not work as one would expect. #3110 explains the issue and how it can be generated.

**Comment by elipe17** [comment]
@reitermb and @victoriaatraft, I made the required updates based on the ACs and was able to get the filter applying logic all hooked into the one `Apply Filters` button. However, I did not implement the state preserving features we discussed because it cannot be done effectively until #3110 is resolved. Finding a path forward on that issue will give us more control over styling and accessibility from TDP.

**Review by raftmsohani — APPROVED** [review]
Please fill out the deliverables section, otherwise looks good to me!

**Comment by reitermb** [comment]
This should be a quick ✅ once I get a look at it, looks like there's currently a deployment issue if ya'll have time to take a look @jtimpe @elipe17

**Review by reitermb — CHANGES_REQUESTED** [review]
One final tweak—could we add an `aria-label="STTs"` to the select control?

e.g. 
image


With that it should be good to go on to QASP! Thanks!

cc @elipe17 @jtimpe

**Comment by elipe17** [comment]
> One final tweak—could we add an `aria-label="STTs"` to the select control?
> 
> e.g. image
> 
> With that it should be good to go on to QASP! Thanks!
> 
> cc @elipe17 @jtimpe

@reitermb got that added. Thank you for the review!

**Review by reitermb — APPROVED** [review]

**Review by ADPennington — CHANGES_REQUESTED** [review]
@elipe17 @reitermb attached is a video of the filter behavior I'm observing in the qasp environment. Please note:

- I cleared my browser caches, cookies, etc. before testing. 
- tried this on Windows Chrome, Windows Edge, and Mac Chrome
- the newest filter appears to not be working as expected. I also thought "newest" was the default?

https://github.com/user-attachments/assets/b6711189-e8c6-462c-8a34-0de9e3a580fe

**Comment by elipe17** [comment]
> @elipe17 @reitermb attached is a video of the filter behavior I'm observing in the qasp environment. Please note:
> 
> * I cleared my browser caches, cookies, etc. before testing.
> * tried this on Windows Chrome, Windows Edge, and Mac Chrome
> * the newest filter appears to not be working as expected. I also thought "newest" was the default?
> 
>  Screen.Recording.2024-08-14.at.8.19.42.AM.mov

@ADPennington very interesting. `newest` should be the default when visiting any of the search indexes pages. As for the rest of the filtering, was your URL reflecting the correct query params? I will try and duplicate this locally in the meantime.

**Comment by ADPennington** [comment]
> > @elipe17 @reitermb attached is a video of the filter behavior I'm observing in the qasp environment. Please note:
> > 
> > * I cleared my browser caches, cookies, etc. before testing.
> > * tried this on Windows Chrome, Windows Edge, and Mac Chrome
> > * the newest filter appears to not be working as expected. I also thought "newest" was the default?
> > 
> > Screen.Recording.2024-08-14.at.8.19.42.AM.mov
> 
> @ADPennington very interesting. `newest` should be the default when visiting any of the search indexes pages. As for the rest of the filtering, was your URL reflecting the correct query params? I will try and duplicate this locally in the meantime.

I believe I was seeing URLs like this:

`https://tdp-frontend-qasp.app.cloud.gov/admin/search_indexes/tanf_t1/?`
`https://tdp-frontend-qasp.app.cloud.gov/admin/search_indexes/tanf_t2/?`

instead of:

`https://tdp-frontend-qasp.app.cloud.gov/admin/search_indexes/tanf_t1/?fiscal_period=2024Q1`
`https://tdp-frontend-qasp.app.cloud.gov/admin/search_indexes/tanf_t2/?fiscal_period=2024Q1`

**Comment by elipe17** [comment]
> > > @elipe17 @reitermb attached is a video of the filter behavior I'm observing in the qasp environment. Please note:
> > > 
> > > * I cleared my browser caches, cookies, etc. before testing.
> > > * tried this on Windows Chrome, Windows Edge, and Mac Chrome
> > > * the newest filter appears to not be working as expected. I also thought "newest" was the default?
> > > 
> > > Screen.Recording.2024-08-14.at.8.19.42.AM.mov
> > 
> > 
> > @ADPennington very interesting. `newest` should be the default when visiting any of the search indexes pages. As for the rest of the filtering, was your URL reflecting the correct query params? I will try and duplicate this locally in the meantime.
> 
> I believe I was seeing URLs like this:
> 
> `https://tdp-frontend-qasp.app.cloud.gov/admin/search_indexes/tanf_t1/?` `https://tdp-frontend-qasp.app.cloud.gov/admin/search_indexes/tanf_t2/?`
> 
> instead of:
> 
> `https://tdp-frontend-qasp.app.cloud.gov/admin/search_indexes/tanf_t1/?fiscal_period=2024Q1` `https://tdp-frontend-qasp.app.cloud.gov/admin/search_indexes/tanf_t2/?fiscal_period=2024Q1`

@ADPennington I pushed the fix for the `newest` filter. It was still filtering correctly, however the widget just wasn't being updated correctly anymore because of a piece of code I had removed. Added that back in and the filter looks and behaves as expected.

**Comment by ADPennington** [comment]
@reitermb @ttran-hub this looks good to me but i did not test the read-out on the first AC. as soon as you have time, can you add the evidence of a11y review here? this branch is currently deployed to [qasp environment](https://tdp-frontend-qasp.app.cloud.gov/home). cc: @elipe17

**Comment by reitermb** [comment]
> @reitermb @ttran-hub this looks good to me but i did not test the read-out on the first AC. as soon as you have time, can you add the evidence of a11y review here? this branch is currently deployed to [qasp environment](https://tdp-frontend-qasp.app.cloud.gov/home). cc: @elipe17


https://github.com/user-attachments/assets/d9e5faf5-13cf-42b3-bf96-53b08e27d339

Added a strikethrough to the second sub-item on the AC since we found a way to replace the extra button with the normal filter button. 

image

We were getting the above invalid aria attribute name error on axe scan (though it wasn't impacting app behavior) but that's corrected with the most recent commit. 

cc @ADPennington

**Review by ADPennington — APPROVED** [review]
🚀

**Pull request merged**